src/System.Management.Automation/engine/LanguagePrimitives.cs

```diff
@@ -3212,6 +3212,56 @@ private static PSConverter<bool> CreateNumericToBoolConverter(Type fromType)
                 valueToConvert.ToString(), resultType.ToString(), exception.Message);
         }
 
+
+        private static Delegate ConvertPSMethodInfoToDelegate(object valueToConvert,
+                                                      Type resultType,
+                                                      bool recurse,
+                                                      PSObject originalValueToConvert,
+                                                      IFormatProvider formatProvider,
+                                                      TypeTable backupTable)
+        {
+            try
+            {
+                var psMethod = (PSMethod)valueToConvert;
+
+                var maybeType = psMethod.instance as Type;
+                var methodInfoCandiates = maybeType != null
+                    ? maybeType.GetMethods(BindingFlags.Static | BindingFlags.Public)
+                    : psMethod.instance.GetType().GetMethods(BindingFlags.Instance | BindingFlags.Public);
+
+                var targetMethodInfo = resultType.GetMethod("Invoke");
+
+                var comparator = new DelegateArgsComparator(targetMethodInfo);
+
+                foreach (var candidate in methodInfoCandiates)
+                {
+                    if (candidate.Name != psMethod.Name)
+                    {
+                        continue;
+                    }
+                    if (comparator.SignatureMatches(candidate.ReturnType, candidate.GetParameters()))
+                    {
+                        return maybeType != null
+                            ? candidate.CreateDelegate(resultType)
+                            : candidate.CreateDelegate(resultType, psMethod.instance);
+                    }
+                }
+                var msg = $"No matching overload found from {psMethod} to {resultType}";
```

Comment:
Strings that appear in error messages need to be in a resource for localization.